ui/inspectors/system-section.reel/language-and-region.reel/language-and-region.js

```diff
@@ -46,12 +107,44 @@ exports.LanguageAndRegion = Component.specialize(/** @lends LanguageAndRegion# *
                 Promise.all(loadingPromises).then(function() {
                     this.isLoading = false;
                 });
+                var today = new Date();
+                this.dateFormatShortOptions = this._generateDateFormatConvertedList(today, this.shortDateFormats);
+                this.dateFormatMediumOptions = this._generateDateFormatConvertedList(today, this.mediumDateFormats);
+                this.dateFormatLongOptions = this._generateDateFormatConvertedList(today, this.longDateFormats);
+                this.dateFormatFullOptions = this._generateDateFormatConvertedList(today, this.fullDateFormats);
+                this.timeFormatShortOptions = this._generateDateFormatConvertedList(today, this.shortTimeFormats);
+                this.timeFormatMediumOptions = this._generateDateFormatConvertedList(today, this.mediumTimeFormats);
+                this.timeFormatLongOptions = this._generateDateFormatConvertedList(today, this.longTimeFormats);
+            }
+        }
+    },
+
+    _generateDateFormatConvertedList: {
+        value: function(today, dateOptionList) {
+            var _formattedDateList = [];
+            for (var i = 0,length = dateOptionList.length; i < length; i++) {
+                this.dateConverter.pattern = dateOptionList[i];
+                _formattedDateList.push({ label: this.dateConverter.convert(today), value: dateOptionList[i] });
             }
+            return _formattedDateList;
+        }
+    },
+
+    _checkUserSettings: {
+        value: function(user){
```

Comment:
it should only take `userSettings` as an argument, it would make the code more readable and it would reflect more the intention.